'use strict';

var pubmedApp = angular.module('pubmedApp', []);

/* http://stackoverflow.com/questions/14478106/angularjs-sorting-by-property */
pubmedApp.filter('orderObjectBy', function(){
	return function(input,attribute)
		{
		if (!angular.isObject(input)) return input;
		
		var array = [];
		for(var objectKey in input) {
			array.push(input[objectKey]);
		    }
		if(attribute == "year")
			{
			array.sort(function(a, b){

				a = a.journal.pubdate.year;
				b = b.journal.pubdate.year;
				
				return a - b;
			    });
			}
		else if(attribute == "authors")
			{
			array.sort(function(a, b){
				a = a.authors.length;
				b = b.authors.length;
				return a - b;
			    });
			}
		else if(attribute == "title")
			{
			array.sort(function(a, b){
				
				a = a.title.toUpperCase();
				b = b.title.toUpperCase();
				return a < b ? -1 : a>b ? 1: 0;
			    });
			}
		else
			{
			
			}
		return array;
		}
	});

pubmedApp.controller('PubmedCtrl',function($scope)
	{
	$scope.orderProp = 'title';
	
	$scope.articles=[
{"pmid":21984761,"title":"Knime4Bio: a set of custom nodes for the interpretation of next-generation sequencing data with KNIME.","journal":{"volume":"27","issue":"22","pubdate":{"year":2011,"month":"Nov","day":15}},"abstract":"Analysing large amounts of data generated by next-generation sequencing (NGS) technologies is difficult for researchers or clinicians without computational skills. They are often compelled to delegate this task to computer biologists working with command line utilities. The availability of easy-to-use tools will become essential with the generalization of NGS in research and diagnosis. It will enable investigators to handle much more of the analysis. Here, we describe Knime4Bio, a set of custom nodes for the KNIME (The Konstanz Information Miner) interactive graphical workbench, for the interpretation of large biological datasets. We demonstrate that this tool can be utilized to quickly retrieve previously published scientific findings.","journalInfo":{"nlmUniqueID":"9808944","medlineTA":"Bioinformatics"},"authors":[{"lastName":"Lindenbaum","foreName":"Pierre","initials":"P","affiliation":"Institut du thorax, Inserm UMR 915, Centre Hospitalier Universitaire de Nantes, 44000 Nantes, France."},{"lastName":"Le Scouarnec","foreName":"Solena","initials":"S"},{"lastName":"Portero","foreName":"Vincent","initials":"V"},{"lastName":"Redon","foreName":"Richard","initials":"R"}]},
{"pmid":22046109,"title":"BioStar: an online question & answer resource for the bioinformatics community.","journal":{"volume":"7","issue":"10","pubdate":{"year":2011,"month":"Oct","day":null}},"abstract":null,"journalInfo":{"nlmUniqueID":"101238922","medlineTA":"PLoS Comput Biol"},"authors":[{"lastName":"Parnell","foreName":"Laurence D","initials":"LD","affiliation":"Nutrition and Genomics Laboratory, Jean Mayer-United States Department of Agriculture Human Nutrition Research Center on Aging at Tufts University, Boston, Massachusetts, United States of America."},{"lastName":"Lindenbaum","foreName":"Pierre","initials":"P"},{"lastName":"Shameer","foreName":"Khader","initials":"K"},{"lastName":"Dall'Olio","foreName":"Giovanni Marco","initials":"GM"},{"lastName":"Swan","foreName":"Daniel C","initials":"DC"},{"lastName":"Jensen","foreName":"Lars Juhl","initials":"LJ"},{"lastName":"Cockell","foreName":"Simon J","initials":"SJ"},{"lastName":"Pedersen","foreName":"Brent S","initials":"BS"},{"lastName":"Mangan","foreName":"Mary E","initials":"ME"},{"lastName":"Miller","foreName":"Christopher A","initials":"CA"},{"lastName":"Albert","foreName":"Istvan","initials":"I"}]},
{"pmid":21378989,"title":"Truncating mutations in the last exon of NOTCH2 cause a rare skeletal disorder with osteoporosis.","journal":{"volume":"43","issue":"4","pubdate":{"year":2011,"month":"Apr","day":null}},"abstract":"Hajdu-Cheney syndrome is a rare autosomal dominant skeletal disorder with facial anomalies, osteoporosis and acro-osteolysis. We sequenced the exomes of six unrelated individuals with this syndrome and identified heterozygous nonsense and frameshift mutations in NOTCH2 in five of them. All mutations cluster to the last coding exon of the gene, suggesting that the mutant mRNA products escape nonsense-mediated decay and that the resulting truncated NOTCH2 proteins act in a gain-of-function manner.","journalInfo":{"nlmUniqueID":"9216904","medlineTA":"Nat Genet"},"authors":[{"lastName":"Isidor","foreName":"Bertrand","initials":"B","affiliation":"CHU Nantes, Service de Génétique Médicale, Nantes, France."},{"lastName":"Lindenbaum","foreName":"Pierre","initials":"P"},{"lastName":"Pichon","foreName":"Olivier","initials":"O"},{"lastName":"Bézieau","foreName":"Stéphane","initials":"S"},{"lastName":"Dina","foreName":"Christian","initials":"C"},{"lastName":"Jacquemont","foreName":"Sébastien","initials":"S"},{"lastName":"Martin-Coignard","foreName":"Dominique","initials":"D"},{"lastName":"Thauvin-Robinet","foreName":"Christel","initials":"C"},{"lastName":"Le Merrer","foreName":"Martine","initials":"M"},{"lastName":"Mandel","foreName":"Jean-Louis","initials":"JL"},{"lastName":"David","foreName":"Albert","initials":"A"},{"lastName":"Faivre","foreName":"Laurence","initials":"L"},{"lastName":"Cormier-Daire","foreName":"Valérie","initials":"V"},{"lastName":"Redon","foreName":"Richard","initials":"R"},{"lastName":"Le Caignec","foreName":"Cédric","initials":"C"}]},
{"pmid":19755503,"title":"The Gene Wiki: community intelligence applied to human gene annotation.","journal":{"volume":"38","issue":"Database issue","pubdate":{"year":2010,"month":"Jan","day":null}},"abstract":"Annotating the function of all human genes is a critical, yet formidable, challenge. Current gene annotation efforts focus on centralized curation resources, but it is increasingly clear that this approach does not scale with the rapid growth of the biomedical literature. The Gene Wiki utilizes an alternative and complementary model based on the principle of community intelligence. Directly integrated within the online encyclopedia, Wikipedia, the goal of this effort is to build a gene-specific review article for every gene in the human genome, where each article is collaboratively written, continuously updated and community reviewed. Previously, we described the creation of Gene Wiki 'stubs' for approximately 9000 human genes. Here, we describe ongoing systematic improvements to these articles to increase their utility. Moreover, we retrospectively examine the community usage and improvement of the Gene Wiki, providing evidence of a critical mass of users and editors. Gene Wiki articles are freely accessible within the Wikipedia web site, and additional links and information are available at http://en.wikipedia.org/wiki/Portal:Gene_Wiki.","journalInfo":{"nlmUniqueID":"0411011","medlineTA":"Nucleic Acids Res"},"authors":[{"lastName":"Huss","foreName":"Jon W","initials":"JW","affiliation":"Genomics Institute of the Novartis Research Foundation, San Diego, CA 92121, USA."},{"lastName":"Lindenbaum","foreName":"Pierre","initials":"P"},{"lastName":"Martone","foreName":"Michael","initials":"M"},{"lastName":"Roberts","foreName":"Donabel","initials":"D"},{"lastName":"Pizarro","foreName":"Angel","initials":"A"},{"lastName":"Valafar","foreName":"Faramarz","initials":"F"},{"lastName":"Hogenesch","foreName":"John B","initials":"JB"},{"lastName":"Su","foreName":"Andrew I","initials":"AI"}]},
{"pmid":19331686,"title":"Robust physical methods that enrich genomic regions identical by descent for linkage studies: confirmation of a locus for osteogenesis imperfecta.","journal":{"volume":"10","issue":null,"pubdate":{"year":2009,"month":null,"day":null}},"abstract":"The monogenic disease osteogenesis imperfecta (OI) is due to single mutations in either of the collagen genes ColA1 or ColA2, but within the same family a given mutation is accompanied by a wide range of disease severity. Although this phenotypic variability implies the existence of modifier gene variants, genome wide scanning of DNA from OI patients has not been reported. Promising genome wide marker-independent physical methods for identifying disease-related loci have lacked robustness for widespread applicability. Therefore we sought to improve these methods and demonstrate their performance to identify known and novel loci relevant to OI.","journalInfo":{"nlmUniqueID":"100966978","medlineTA":"BMC Genet"},"authors":[{"lastName":"Brooks","foreName":"Peter","initials":"P","affiliation":"IntegraGen SA, Evry, France. peter.brooks@parisbc.com"},{"lastName":"Marcaillou","foreName":"Charles","initials":"C"},{"lastName":"Vanpeene","foreName":"Maud","initials":"M"},{"lastName":"Saraiva","foreName":"Jean-Paul","initials":"JP"},{"lastName":"Stockholm","foreName":"Daniel","initials":"D"},{"lastName":"Francke","foreName":"Stephan","initials":"S"},{"lastName":"Favis","foreName":"Reyna","initials":"R"},{"lastName":"Cohen","foreName":"Nadine","initials":"N"},{"lastName":"Rousseau","foreName":"Francis","initials":"F"},{"lastName":"Tores","foreName":"Frédéric","initials":"F"},{"lastName":"Lindenbaum","foreName":"Pierre","initials":"P"},{"lastName":"Hager","foreName":"Jörg","initials":"J"},{"lastName":"Philippi","foreName":"Anne","initials":"A"}]},
{"pmid":18053270,"title":"Association of autism with polymorphisms in the paired-like homeodomain transcription factor 1 (PITX1) on chromosome 5q31: a candidate gene analysis.","journal":{"volume":"8","issue":null,"pubdate":{"year":2007,"month":null,"day":null}},"abstract":"Autism is a complex, heterogeneous, behaviorally-defined disorder characterized by disruptions of the nervous system and of other systems such as the pituitary-hypothalamic axis. In a previous genome wide screen, we reported linkage of autism with a 1.2 Megabase interval on chromosome 5q31. For the current study, we hypothesized that 3 of the genes in this region could be involved in the development of autism: 1) paired-like homeodomain transcription factor 1 (PITX1), which is a key regulator of hormones within the pituitary-hypothalamic axis, 2) neurogenin 1, a transcription factor involved in neurogenesis, and 3) histone family member Y (H2AFY), which is involved in X-chromosome inactivation in females and could explain the 4:1 male:female gender distortion present in autism.","journalInfo":{"nlmUniqueID":"100968552","medlineTA":"BMC Med Genet"},"authors":[{"lastName":"Philippi","foreName":"Anne","initials":"A","affiliation":"IntegraGen SA, 5 rue Henri Desbruères, Genopole Campus 1, Genavenir 8, 91000 Evry, France. anne.philippi@integragen.com"},{"lastName":"Tores","foreName":"Frédéric","initials":"F"},{"lastName":"Carayol","foreName":"Jérome","initials":"J"},{"lastName":"Rousseau","foreName":"Francis","initials":"F"},{"lastName":"Letexier","foreName":"Mélanie","initials":"M"},{"lastName":"Roschmann","foreName":"Elke","initials":"E"},{"lastName":"Lindenbaum","foreName":"Pierre","initials":"P"},{"lastName":"Benajjou","foreName":"Abdel","initials":"A"},{"lastName":"Fontaine","foreName":"Karine","initials":"K"},{"lastName":"Vazart","foreName":"Céline","initials":"C"},{"lastName":"Gesnouin","foreName":"Philippe","initials":"P"},{"lastName":"Brooks","foreName":"Peter","initials":"P"},{"lastName":"Hager","foreName":"Jörg","initials":"J"}]},{"pmid":16027742,"title":"Haplotypes in the gene encoding protein kinase c-beta (PRKCB1) on chromosome 16 are associated with autism.","journal":{"volume":"10","issue":"10","pubdate":{"year":2005,"month":"Oct","day":null}},"abstract":"Autism is a developmental disorder characterized by impairments in social interaction and communication associated with repetitive patterns of interest or behavior. Autism is highly influenced by genetic factors. Genome-wide linkage and candidate gene association approaches have been used to try and identify autism genes. A few loci have repeatedly been reported linked to autism. Several groups reported evidence for linkage to a region on chromosome 16p. We have applied a direct physical identity-by-descent (IBD) mapping approach to perform a high-density (0.85 megabases) genome-wide linkage scan in 116 families from the AGRE collection. Our results confirm linkage to a region on chromosome 16p with autism. High-resolution single-nucleotide polymorphism (SNP) genotyping and analysis of this region show that haplotypes in the protein kinase c-beta gene are strongly associated with autism. An independent replication of the association in a second set of 167 trio families with autism confirmed our initial findings. Overall, our data provide evidence that the PRKCB1 gene on chromosome 16p may be involved in the etiology of autism.","journalInfo":{"nlmUniqueID":"9607835","medlineTA":"Mol Psychiatry"},"authors":[{"lastName":"Philippi","foreName":"A","initials":"A","affiliation":"IntegraGen SA, Genopole, Evry, France."},{"lastName":"Roschmann","foreName":"E","initials":"E"},{"lastName":"Tores","foreName":"F","initials":"F"},{"lastName":"Lindenbaum","foreName":"P","initials":"P"},{"lastName":"Benajou","foreName":"A","initials":"A"},{"lastName":"Germain-Leclerc","foreName":"L","initials":"L"},{"lastName":"Marcaillou","foreName":"C","initials":"C"},{"lastName":"Fontaine","foreName":"K","initials":"K"},{"lastName":"Vanpeene","foreName":"M","initials":"M"},{"lastName":"Roy","foreName":"S","initials":"S"},{"lastName":"Maillard","foreName":"S","initials":"S"},{"lastName":"Decaulne","foreName":"V","initials":"V"},{"lastName":"Saraiva","foreName":"J P","initials":"JP"},{"lastName":"Brooks","foreName":"P","initials":"P"},{"lastName":"Rousseau","foreName":"F","initials":"F"},{"lastName":"Hager","foreName":"J","initials":"J"}]},
{"pmid":9682060,"title":"CloneIt: finding cloning strategies, in-frame deletions and frameshifts.","journal":{"volume":"14","issue":"5","pubdate":{"year":1998,"month":"Jun","day":null}},"abstract":"The CloneIt program searches for sub-cloning strategies, in-frame deletions and frameshifts within a plasmid sequence.","journalInfo":{"nlmUniqueID":"9808944","medlineTA":"Bioinformatics"},"authors":[{"lastName":"Lindenbaum","foreName":"P","initials":"P","affiliation":"Laboratoire de Biologie Moléculaire des Rotavirus, Virologie et Immunologie Moléculaires, Institut National de la Recherche Agronomique, 78350 Jouy-en-Josas Cedex, France. lindenb@biotec.jouy.inra.fr"}]},
{"pmid":8985320,"title":"In vivo and in vitro phosphorylation of rotavirus NSP5 correlates with its localization in viroplasms.","journal":{"volume":"71","issue":"1","pubdate":{"year":1997,"month":"Jan","day":null}},"abstract":"NSP5 (NS26), the product of rotavirus gene 11, is a phosphoprotein whose role in the virus replication cycle is unknown. To gain further insight into its function, we obtained monoclonal antibodies against the baculovirus-expressed protein. By immunoprecipitation and immunoblotting experiments, we showed that (i) NSP5 appears in many different phosphorylated forms in rotavirus-infected cells; (ii) immunoprecipitated NSP5 from rotavirus-infected cells can be phosphorylated in vitro by incubation with ATP; (iii) NSP5, produced either by transient transfection of rotavirus gene 11 or by infection by gene 11 recombinant vaccinia virus or baculovirus, can be phosphorylated in vivo and in vitro; (iv) NSP5 expressed in Escherichia coli is phosphorylated in vitro, and thus NSP5 is a potential protein kinase; and (v) NSP5 forms dimers and interacts with NSP2. The intracellular localization of NSP5 in the course of rotavirus infection and after transient expression in COS7 cells has also been investigated. In rotavirus-infected cells, NSP5 is localized in viroplasms, but it is widespread throughout the cytoplasm of transfected COS7 cells. NSP5 produced by transfected COS7 cells did not acquire the multiphosphorylated forms observed in rotavirus-infected COS7 cells. Thus, there is a tight correlation between the localization of NSP5 in the viroplasms and its protein kinase activity in vivo or in vitro. Our results suggest that cellular or viral cofactors are indispensable to fully phosphorylate NSP5 and to reach its intracellular localization.","journalInfo":{"nlmUniqueID":"0113724","medlineTA":"J Virol"},"authors":[{"lastName":"Poncet","foreName":"D","initials":"D","affiliation":"Laboratoire de Virologie et Immunologie Moléculaires INRA, CRJJ, Jouy-en-Josas, France. poncet@biotec.jouy.inra.fr"},{"lastName":"Lindenbaum","foreName":"P","initials":"P"},{"lastName":"L'Haridon","foreName":"R","initials":"R"},{"lastName":"Cohen","foreName":"J","initials":"J"}]}
	];
	});
